folding-schemes/src/arith/r1cs.rs

```diff
@@ -92,6 +97,70 @@ impl<F: PrimeField> RelaxedR1CS<F> {
 
         Ok(())
     }
+
+    // Computes the E term, given A, B, C, z, u
+    fn compute_E(
+        A: &SparseMatrix<F>,
+        B: &SparseMatrix<F>,
+        C: &SparseMatrix<F>,
+        z: &[F],
+        u: &F,
+    ) -> Result<Vec<F>, Error> {
+        let Az = mat_vec_mul(A, z)?;
+        let Bz = mat_vec_mul(B, z)?;
+        let AzBz = hadamard(&Az, &Bz)?;
+
+        let Cz = mat_vec_mul(C, z)?;
+        let uCz = vec_scalar_mul(&Cz, u);
+        vec_sub(&AzBz, &uCz)
+    }
```

Comment:
Interesting, I agree! Maybe we can do this in a different PR? @arnaucube might also have a take on this.

Reply:
Sure!!! We can simply fill an issue and take benchmarks. Or leave it for anyone that wants to give it a shot!

Reply:
Regarding parallelization, I started some months ago the branch 'parallelize', which a couple of days ago I ported to latest changes of the repo, to experiment with parallelization of these kinds of things (eg. matrix/vector operations, the computation of the crossterms, the commitments to the crossterms, etc)
can be found here: https://github.com/privacy-scaling-explorations/sonobe/commit/096e2f358778a4901020555f7ae632e4e043247e
But didn't open a PR because I think that is better to wait a bit more, because maybe we are better parallelizing at higher levels rather than at internals (or maybe not, I'm not an expert on parallelization).